### PR TITLE
Fix bugs for #3943

### PR DESCRIPTION
**Pull request by alexyaoyang**
## Improvements:
- Changed references of walk to walk_min in follow path and spiral workers.
- Swap walk_min and walk_max if walk_min > walk_max

### CODE REVIEW

File: pokemongo_bot/cell_workers/follow_path.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def work(self):
             lng
         )
 
-        if dist <= 1 or (self.bot.config.walk > 0 and is_at_destination):
+        if dist <= 1 or (self.bot.config.walk_max > 0 and is_at_destination):
```
Comment (BriceSD) [inline pokemongo_bot/cell_workers/follow_path.py]: Why ```walk_max``` and not ```walk_min``` ? 
Having both seems more appropriate to me : ```self.bot.config.walk_max > 0 && self.bot.config.walk_min > 0```

File: pokemongo_bot/cell_workers/follow_spiral.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def work(self):
                 }
             )
 
-        if dist <= 1 or (self.bot.config.walk > 0 and step_walker == None):
+        if dist <= 1 or (self.bot.config.walk_max > 0 and step_walker == None):
```
Comment (BriceSD) [inline pokemongo_bot/cell_workers/follow_spiral.py]: Why ```walk_max``` and not ```walk_min``` ? 
Having both seems more appropriate to me : ```self.bot.config.walk_max > 0 && self.bot.config.walk_min > 0```

**Comment by BriceSD** [comment]
Why ```walk_max``` and not ```walk_min``` ? 
Having both seems more appropriate to me : ```self.bot.config.walk_max > 0 && self.bot.config.walk_min > 0```

**Comment by BriceSD** [comment]
@mjmadsen Please review, need to fix asap.

**Comment by Setsu-BHMT** [comment]
The intention of the original code was to check if the bot was supposed to walk at all; given that interpretation it might be more appropriate to check walk_min, though it's probably a good idea to rethink the logic here. Checking both shouldn't be necessary as logically if walk_min > 0 then walk_max must be >0 or ==0; this is assuming of course that we have a check somewhere to make sure walk_max is greater than or equal to walk_min (do we have this?)

**Comment by alexyaoyang** [comment]
Yup, from what I can see, those if condition checks if user's config allows the bot to walk at all.

Thus checking for one of them should be sufficient.

@Setsu-BHMT You are right, I just tried, we do not have that check in place. I'll add a swap if min > max.

**Comment by BriceSD** [comment]
#4021

**Comment by douglascamata** [comment]
The bot should not support teleporting anymore, as it will result is super fast ban. We should only support walking "normally", so those checks can be entirely removed, imho.

**Comment by alexyaoyang** [comment]
@douglascamata I think checking if walking speed is > 0 is not related to teleporting?

**Comment by douglascamata** [comment]
@alexyaoyang walk speed should be required and bot should use the default value of 4.16 if it's not provided in the config. If walk speed is 0, we need to bump it up to 4.16.

**Comment by alexyaoyang** [comment]
@douglascamata I'm not sure if we are on the same issue here. 

One issue is checking if walk_min is > 0 in the respective follow walkers, I would recommend this condition check to stay put. 

Another issue is to disable teleporting, which I think is not related to this PR.

Final issue is bumping of walking speed of 0, which I have just committed. I took the liberty to randomize the bumped up walking speed between [3.0,5.0)

**Comment by mjmadsen** [comment]
#4046 Didn't see this PR before posting. Either PR should suffice.

**Comment by TheSavior** [comment]
I merged #4046. I'm really surprised this one doesn't look like it has merge conflicts. I'll let you close this one if you feel like the other was sufficient.

**Comment by alexyaoyang** [comment]
Sure, closing this PR right now.

**Comment by CyberMew** [comment]
Maybe some of the changes not present in the other PR could be picked over.

**Comment by alexyaoyang** [comment]
@CyberMew I'll create a new PR for the changes not present in #4046.

**Pull request closed without merging**